esinstall/src/entrypoints.ts

```diff
@@ -0,0 +1,268 @@
+        browserField['./index.js'] ||
+        browserField['./index'] ||
+        browserField['index.js'] ||
+        browserField['index'] ||
+        browserField['./'] ||
+        browserField['.'];
+
+      if (typeof browserEntrypoint === 'string') {
+        foundEntrypoint = browserEntrypoint;
+      }
+    }
+  }
+
+  // If browser object is set but no relevant entrypoint is found, fall back to "main".
+  if (!foundEntrypoint) {
+    foundEntrypoint = manifest.main;
+  }
+
+  return foundEntrypoint;
+}
+
+export function resolveMainEntrypoint(manifest: PackageManifest) {
+  return resolveManifestEntry(manifest);
+}
+
+/**
+ * Given an ExportMapEntry find the entry point, resolving recursively.
+ */
+export function resolveExportMapEntry(
+  exportMapEntry?: ExportMapEntry,
+  conditions?: string[],
+): string | undefined {
+  // If this is a string or undefined we can skip checking for conditions
+  switch (typeof exportMapEntry) {
+    case 'string':
+      return exportMapEntry;
+    case 'undefined':
+      return exportMapEntry;
+  }
+
+  let entry = exportMapEntry;
+  for (let i = 0, len = conditions?.length || 0; i < len; i++) {
+    let condition = conditions![i];
+
+    if (entry[condition]) {
+      entry = entry[condition];
+    }
+  }
+
+  return (
+    resolveExportMapEntry(entry?.browser) ||
+    resolveExportMapEntry(entry?.import) ||
+    resolveExportMapEntry(entry?.default) ||
+    resolveExportMapEntry(entry?.require) ||
+    undefined
+  );
+}
+
+function resolveExportsMap(
+  packageManifest: PackageManifestWithExports,
+  exportsProp: string,
+): string | undefined {
+  const exportMapEntry = packageManifest.exports[exportsProp];
+  return resolveExportMapEntry(exportMapEntry);
+}
+
+type ResolveManifestFn = (dep: string, cwd: string) => [string | null, any];
+
+type ResolveEntrypointOptions = {
+  cwd: string;
+  packageLookupFields: string[];
+  resolveManifest?: ResolveManifestFn | undefined;
+};
+
+/**
+ * Resolve a "webDependencies" input value to the correct absolute file location.
+ * Supports both npm package names, and file paths relative to the node_modules directory.
+ * Follows logic similar to Node's resolution logic, but using a package.json's ESM "module"
+ * field instead of the CJS "main" field.
+ */
+export function resolveEntrypoint(
+  dep: string,
+  {cwd, packageLookupFields}: ResolveEntrypointOptions,
+): string {
+  // We first need to check for an export map in the package.json. If one exists, resolve to it.
+  const [packageName, packageEntrypoint] = parsePackageImportSpecifier(dep);
+  const [packageManifestLoc, packageManifest] = resolveDependencyManifest(packageName, cwd);
+
+  if (packageManifestLoc && packageManifest && typeof packageManifest.exports !== 'undefined') {
+    // If this is a non-main entry point
+    if (packageEntrypoint) {
+      const exportMapValue = resolveExportsMap(
+        packageManifest as PackageManifestWithExports,
+        './' + packageEntrypoint,
```

Comment:
I could see it being very easy for us to make a mistake and give `resolveExportsMap()` the argument `packageEntrypoint` instead of `./ + packageEntrypoint`. Can this function either error when doesn't start with `"./"` or handle the import without the prefix?

Reply:
I got rid of this function as it didn't really add much.